### ISSUE TITLE
Delete Button

### ISSUE THREAD

**Issue by jameskerr** [brimdata/zui#2768]
A button that could call out to the delete api would be a handy feature. It could work like this.

Run a query.
Click a delete button.
Confirm.
Execute.

**Comment by philrz** [comment]
I agree this would be a great feature! Two subtleties I wanted to point out:

1. Per the [limitations](https://zed.brimdata.io/docs/commands/zed#24-delete), to be used with `delete` the query has to be a single filter expression. Hopefully the app could sniff out when a user's query falls into that criteria so it can provide some UX hint to the user when delete is possible (and, when it isn't, maybe provide some kind of link off to the docs that explain that so they can make the necessary adjustments.)

2. Since the deletion is based on a filter, it's theoretically possible that the data in the pool might change between when the initial query is executed & results are shown for review and when the delete button is clicked and the deletion actually occurs. Of course this same exposure would be there if a user did a la carte `zed query` and `zed delete` operations, but since the app will make it look so easy & seamless, maybe there's a way we could help limit the exposure, perhaps with some help from the backend. This is just one random thought, but maybe there could be an option on the Zed lake query API endpoint to request a hash of query results, then provide a similar hash of what-was-deleted when deleting, and so the app could compare the two and when they match, great... and when not, it could point the user to some docs that explain how to use the commit log to review the pool's history and see if they're ok with what got deleted & roll back if not.

**Comment by jameskerr** [comment]
Yes I agree here Phil on both points.
